topdown/eval.go

```diff
@@ -2781,7 +2781,9 @@ func merge(a, b ast.Value) (ast.Value, bool) {
 	if ok1 && ok2 {
 		return mergeObjects(aObj, bObj)
 	}
-	return nil, false
+
+	// nothing to merge, a wins (the mock data)
```

Comment:
I noticed this could be called from `evalTree.extent()` as well (~L1761 is `merged, ok := merge(base, virtual)`) where we merge the result of computing the virtual document for some path with any base document data. Now the base document data will win (which is consistent with existing tests AFAIK.) 👍

Reply:
nit: I'd remove the parenthetical because this applies to merging of base and virtual docs as part of evalTree as well as values set by `with`.